pyanaconda/ui/tui/tuiobject.py

```diff
@@ -55,3 +87,169 @@ def has_help(self):
     def refresh(self, args=None):
         """Put everything to display into self.window list."""
         UIScreen.refresh(self, args)
+
+
+class Dialog(object):
+
+    def __init__(self, title, conditions=None, report_func=reporting_callback):
```

Comment:
Maybe it is not necessary to have `report_func` in `__init__` since it is unlikely to ever change it in anaconda.

Reply:
I would leave it here. When it has default value there is no need to remove it and it could be handy in the future.

Reply:
Ok.